addons/website_forum/views/website_forum.xml

```diff
@@ -1486,7 +1486,7 @@
                                         <t t-set="inDropdown" t-value="True"/>
                                         <t t-set="icon" t-value="'fa-trash-o text-muted'"/>
                                         <t t-set="classes" t-value="'comment_delete'"/>
-                                        <t t-set="karma" t-value="not object.can_unlink and object.karma_unlink or 0"/>
+                                        <t t-set="karma" t-value="not (user._is_admin() or user.karma >= unlink_comment_required_karma) and unlink_comment_required_karma or 0"/>
```

Comment:
From what I see in the code, condition is only based on required karma:
https://github.com/odoo/odoo/blob/15.0/addons/website_forum/models/forum.py#L901

And not related to admin privileges?

Reply:
The `object.can_unlink` is equal to `is_admin or user.karma >= post.karma_unlink`.
As it is a fix in stable I just rebuild the same condition as this one but relative to the comment and not the post.
Also in the [python](https://github.com/odoo/odoo/blob/15.0/addons/website_forum/models/forum.py#L492-L506), every action is authorized as soon as the user is an admin.

However I agree that the `is_admin` is strange here as there won't be any warning/the button won't be muted as soon as the user is an admin.
The action will still be blocked in the python (not the js) if the admin doesn't have enough karma though:
![image](https://github.com/odoo/odoo/assets/107679666/06441647-516e-4496-b167-ed27679e92eb)

I think the admin privilege here are maybe involved as a try to block the action if a certain amount of karma is required and the user is public. I say that looking at the karma required js condition [here](https://github.com/odoo/odoo/blob/15.0/addons/website_forum/static/src/js/website_forum.js#L254).

Should I keep it this way in stable and update the condition to something like this in master ?:
```python
unlink_comment_required_karma
if unlink_comment_required_karma and (user._is_public or user.karma < unlink_comment_required_karma)
else
0
```
If the user doesn't have enough karma, the cool js warning will be shown and we'll still have the python check just to be sure? :thinking: 

We'll maybe need to change every link_button template karma condition then as this `is_admin` check is used everywhere [here](https://github.com/odoo/odoo/blob/15.0/addons/website_forum/views/website_forum.xml#L1219-L1261) (all the `can_...` fields are using the `is_admin`)


-----------------------------------------------------------------------------------------------------------------------

Aside of that I tried to set the karma required to 0 (meaning the action can be performed by every user) and do the action as public user but the rpc is crashing :thinking: ... Not exactly sure why?
![image](https://github.com/odoo/odoo/assets/107679666/001a5b26-137d-4959-b777-8cb178959069)

All in all it's kind of a mess :sweat_smile: 
Sorry for this long comment :sweat_smile:

Reply:
Hello @amdi-odoo 

Seems complicated indeed :smile: 

I see what you mean now, admin is indeed taken into account for others "can_...".
Then, to match a bit more what is done in the forum XML, we could maybe do:

```xml
<t t-set="unlink_comment_required_karma" t-value="message.author_id.id == user.partner_id.id and object.forum_id.karma_comment_unlink_own or object.forum_id.karma_comment_unlink_all"/>
<t t-set="can_unlink_comment" t-value="user._is_admin() or user.karma >= unlink_comment_required_karma"/>
...
<t t-set="karma" t-value="unlink_comment_required_karma if not can_unlink_comment else 0"/>
```

That seems like it would take into account everything and be more inline with others' styling?

About the fact that it doesn't work for public users, could it be:
https://github.com/odoo/odoo/blob/15.0/addons/website_forum/controllers/main.py#L705 ?

Thanks for the explanations!

Reply:
Hello again @awa-odoo :slightly_smiling_face: 
Maybe more something like so for the condition:
```xml
<t t-set="karma" t-value="unlink_comment_required_karma if not can_unlink_comment and unlink_comment_required_karma else 0"/>
```
I added `and unlink_comment_required_karma` as the previous condition was:
`a and b or c` which is the same as: `b if a and b else c` (if I get it right)


I also need to add `:not(.karma_required)` to the delete comment selector:
`'click .comment_delete:not(.karma_required)': '_onDeleteCommentClick',`
to prevent the python rpc from executing when there is a js warning or else we'll have a double alert when the user doesn't have enough karma:
![image](https://github.com/odoo/odoo/assets/107679666/0df2c623-731d-4818-acd8-8f4ac50a0455)


Regarding the rpc error, I set a breakpoint in the js on the rpc calling line and a breakpoint in the python on the first line of the route.
The js breakpoint is triggered but never the python one. The route doesn't seem to be called when the user is public even though the user has enought karma, it just throws the warning `Your Odoo session expired` that I set above  :thinking: 

Thanks for your help!

Reply:
Seems fine for the code changes :slightly_smiling_face: 

As for the explanation of the issue, I was a bit hasty when checking, it's because the auth method is set as "user", preventing public users from calling it altogether:
https://github.com/odoo/odoo/blob/15.0/addons/website_forum/controllers/main.py#L702

Reply:
@awa-odoo :open_mouth: Indeed! Thanks!
So we're probably good here ? :smile:

I don't think I can change the `auth="users"` to `auth="public"` in stable so I'll create a master task to handle it correctly :slightly_smiling_face: (set it to auth="public" with js error handling like the post flag action [[1]](https://github.com/odoo/odoo/blob/master/addons/website_forum/controllers/website_forum.py#L609) [[2]](https://github.com/odoo/odoo/blob/master/addons/website_forum/static/src/js/website_forum.js#L350-L357) )

Thanks for your help!

Reply:
Note that I'm not 100% sure it's the right move.

If these actions can't be called by a public user, then the method can (and should) stay like this.
If we want "pretty" error management, I'm sure there is a way to check that the user is public from the JS somehow :smile: 

Cheers!